From: Simon McVittie <smcv@debian.org>
Date: Fri, 26 Jul 2019 23:51:39 +0100
Subject: gvariant test: Don't run at build-time on mips

DEB_ALLOW_FLAKY_TESTS is not quite right here, because we don't know
that the test would fail if left for long enough - the problem is that
it doesn't get there, because generating random floating-point numbers
is very slow on some of our mips hardware. However, it has the right
practical effect.

Signed-off-by: Simon McVittie <smcv@debian.org>
Bug: https://bugs.debian.org/932678
Forwarded: no
---
 glib/tests/gvariant.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/glib/tests/gvariant.c b/glib/tests/gvariant.c
index 3905e20..d418c75 100644
--- a/glib/tests/gvariant.c
+++ b/glib/tests/gvariant.c
@@ -2405,6 +2405,14 @@ test_fuzzes (gpointer data)
   gdouble fuzziness;
   int i;
 
+#ifdef __mips__
+  if (g_getenv ("DEB_ALLOW_FLAKY_TESTS") == NULL)
+    {
+      g_test_skip ("Extremely slow on some mips CPUs: #932678");
+      return;
+    }
+#endif
+
   fuzziness = GPOINTER_TO_INT (data) / 100.;
 
   for (i = 0; i < 200; i++)
